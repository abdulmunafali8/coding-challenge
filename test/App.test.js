import React from 'react';
import ReactDOM from 'react-dom';
import App from '../src/components/App';
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import { Switch, Route, Redirect } from 'react-router-dom'


Enzyme.configure({ adapter: new Adapter() })

describe('App', () => {

  it('should be defined', () => {
    expect(App).toBeDefined();
  })

  it('should render correctly', () => {
    const tree = shallow(
      <App name='button test' />
    )

    expect(tree).toMatchSnapshot()
  })

})

describe('Route', () => {
  it('should be defined', () => {
    expect(Route).toBeDefined();
  })
})

describe('Redirect', () => {
  it('should be defined', () => {
    expect(Redirect).toBeDefined();
  })
})
