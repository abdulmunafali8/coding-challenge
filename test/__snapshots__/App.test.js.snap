// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    name="button test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <withRouter(Header) />,
        <div
          className="ph3 pv1 background-gray"
        >
          <Switch>
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />
            <Route
              component={[Function]}
              exact={true}
              path="/create"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/login"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/search"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/top"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/report/:id"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/new/:page"
            />
          </Switch>
        </div>,
      ],
      "className": "center w85",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Switch>
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />
            <Route
              component={[Function]}
              exact={true}
              path="/create"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/login"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/search"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/top"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/report/:id"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/new/:page"
            />
          </Switch>,
          "className": "ph3 pv1 background-gray",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                exact={true}
                path="/"
                render={[Function]}
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/create"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/login"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/search"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/top"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/report/:id"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/new/:page"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/create",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/login",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/search",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/top",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/report/:id",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/new/:page",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withRouter(Header) />,
          <div
            className="ph3 pv1 background-gray"
          >
            <Switch>
              <Route
                exact={true}
                path="/"
                render={[Function]}
              />
              <Route
                component={[Function]}
                exact={true}
                path="/create"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/login"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/search"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/top"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/report/:id"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/new/:page"
              />
            </Switch>
          </div>,
        ],
        "className": "center w85",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Switch>
              <Route
                exact={true}
                path="/"
                render={[Function]}
              />
              <Route
                component={[Function]}
                exact={true}
                path="/create"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/login"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/search"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/top"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/report/:id"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/new/:page"
              />
            </Switch>,
            "className": "ph3 pv1 background-gray",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  exact={true}
                  path="/"
                  render={[Function]}
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/create"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/login"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/search"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/top"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/report/:id"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/new/:page"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "exact": true,
                  "path": "/",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/create",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/login",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/search",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/top",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/report/:id",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/new/:page",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Route should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    name="button test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <withRouter(Header) />,
        <div
          className="ph3 pv1 background-gray"
        >
          <Switch>
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />
            <Route
              component={[Function]}
              exact={true}
              path="/create"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/login"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/search"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/top"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/report/:id"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/new/:page"
            />
          </Switch>
        </div>,
      ],
      "className": "center w85",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Switch>
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />
            <Route
              component={[Function]}
              exact={true}
              path="/create"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/login"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/search"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/top"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/report/:id"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/new/:page"
            />
          </Switch>,
          "className": "ph3 pv1 background-gray",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                exact={true}
                path="/"
                render={[Function]}
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/create"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/login"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/search"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/top"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/report/:id"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/new/:page"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/create",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/login",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/search",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/top",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/report/:id",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/new/:page",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withRouter(Header) />,
          <div
            className="ph3 pv1 background-gray"
          >
            <Switch>
              <Route
                exact={true}
                path="/"
                render={[Function]}
              />
              <Route
                component={[Function]}
                exact={true}
                path="/create"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/login"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/search"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/top"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/report/:id"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/new/:page"
              />
            </Switch>
          </div>,
        ],
        "className": "center w85",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Switch>
              <Route
                exact={true}
                path="/"
                render={[Function]}
              />
              <Route
                component={[Function]}
                exact={true}
                path="/create"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/login"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/search"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/top"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/report/:id"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/new/:page"
              />
            </Switch>,
            "className": "ph3 pv1 background-gray",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  exact={true}
                  path="/"
                  render={[Function]}
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/create"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/login"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/search"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/top"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/report/:id"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/new/:page"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "exact": true,
                  "path": "/",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/create",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/login",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/search",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/top",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/report/:id",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/new/:page",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
